src/slackbot.js

```diff
@@ -35,147 +13,255 @@ let data = {
 // - language: language of the code
 // - commitNum: commit # of pull request
 exports.newPR = (pName, prNum, cEmail, rEmails, code, language, commitNum) => {
-	// create a channel
-	request.post('https://slack.com/api/groups.create', {
-		json: {
-		  token: SLACK_TOKEN,
-		  // name of channel = name of project - pull request #
-		  name: `${pName} - ${prNum}`
+	var data = {
+		channelId: '',
+		channelName: '',
+		contributorId: '',
+		contributorName: ''
+	};
+
+	// create channel
+	axios.post('https://slack.com/api/groups.create', {
+		token: SLACK_TOKEN,
+		// name of channel = name of project - pull request #
+		name: `${pName} - ${prNum}`
+	})
+	.then((res) => {
+  		console.log(`statusCode: ${res.statusCode}`);
+		// log result
+		console.log(res.body.ok);
+		// request failed
+		if (res.body.ok === false) {
+			// log error
+			console.log(res.body.error);
+			return;
 		}
-	  }, (error, res, body) => {
-		if (error) {
-		  console.error(error)
-		  return;
+		// success
+		// save channel info
+		data.channelId = res.body.group.id;
+		data.channelName = res.body.group.name;
+		
+		// get users list 
+		return axios.get('https://slack.com/api/users.list', {
+			params: {
+			  token: SLACK_TOKEN,			  
+			}
+		});
+	})
+	.then((res) => {
+		// log result
+		console.log(res.body.ok);
+		// request failed
+		if (res.body.ok === false) {
+			// log error
+			console.log(res.body.error);
+			return;
 		}
-		console.log(`statusCode: ${res.statusCode}`);
-		console.log(body.ok);
-		if (body.ok === false) {
-			// if create channel fails
-			console.log(body.error);
+		// success
+		// loop the users list to find users with the given emails
+		var ids = [];
```

Comment:
Consider use the build-in JS array method to accomplish this task. Nothing wrong with looping it over but I think this way it's cleaner.
```
let ids = res.body.members
		.filter(m => {
			if (m.profile.email === cEmail) {
				data.contributorId = m.id;
				data.contributorName = m.name;
				return true;
			} else {
				return rEmails.includes(m.profile.email);
			}
		})
		.map(m => m.profile.ids);
```